aquadoggo/src/materializer/tasks/reduce.rs

```diff
@@ -1,17 +1,270 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 
 use p2panda_rs::document::DocumentBuilder;
+use p2panda_rs::storage_provider::traits::OperationStore;
 
 use crate::context::Context;
-use crate::materializer::worker::{TaskError, TaskResult};
+use crate::db::traits::DocumentStore;
+use crate::materializer::worker::{Task, TaskError, TaskResult};
 use crate::materializer::TaskInput;
 
-pub async fn reduce_task(_context: Context, _input: TaskInput) -> TaskResult<TaskInput> {
-    // @TODO: Load operations from database
-    let _document = DocumentBuilder::new(vec![])
-        .build()
-        .map_err(|_| TaskError::Failure)?;
+pub async fn reduce_task(context: Context, input: TaskInput) -> TaskResult<TaskInput> {
```

Comment:
Control flow in this function is quite deeply nested. So, if you want to (😉) you could factor out the logic for document id vs. document view id inputs into two separate functions to flatten it out a bit and improve code quality. However, maybe you also want to move on, and let it be as it is.

A short description of what a reduce task actually does would be nice as a docstring.